core/controllers/topics_and_skills_dashboard.py

```diff
@@ -296,16 +296,61 @@ def post(self):
 class NewTopicHandler(base.BaseHandler):
     """Creates a new topic."""
 
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'POST': {
+            'name': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'is_valid_username_string'
+                    }]
+                }
+            },
+            'url_fragment': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'description': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'filename': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'thumbnailBgColor': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'image': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'page_title_fragment': {
+                'schema': {
+                    'type': 'basestring'
+                }
```

Comment:
Should there be more rules for these?

Reply:
Thanks for the input @vojtechjelinek! Should the validators correspond to the try-except clauses further down?

Reply:
For what I can see in the [docs](https://github.com/oppia/oppia/wiki/Schemas#extra-validators), I should validate filename and image using the validators in the class `_Validators` in `schema_utils.py`, is this correct? I cannot find any validator in that file that correspond to what I want to validate, could you help me? @vojtechjelinek

Reply:
Yep replacing try-except clauses is a good way to start. You can look for other files where we validate similar things as a blueprint too.